### PR TITLE
Adding DOCTYPE parse errors

### PR DESCRIPTION
**Pull request by diervo**
/cc @inikulin

### CODE REVIEW

**Comment by inikulin** [comment]
@diervo I'll check it tomorrow if you wouldn't mind. Lot's of thing to do at the moment.

File: source
Comment (inikulin) [inline source]: It should be pars**e**-error-abrupt-closing-of-doctype. You also will need to replace `dfn` with `span` in `Before DOCTYPE public identifier state` for this error.

File: source
Comment (inikulin) [inline source]: Sorry for the wrong suggestion, we can reuse `missing-quote-before-doctype-public-identifier` error here.

File: source
Comment (inikulin) [inline source]: Missing space before `parse error`.

File: source
Comment (inikulin) [inline source]: Should be same as below: `missing-whitespace-between-doctype-public-and-system-identifiers`. I guess it's caused by my misleading comment again.

File: source
Comment (inikulin) [inline source]: Missing space before `parse error`.

File: source
Comment (inikulin) [inline source]: Missing space before `parse error`.

File: source
Comment (inikulin) [inline source]: Missing `<span>parse error</span>`

**Review by inikulin — CHANGES_REQUESTED** [review]

**Comment by diervo** [comment]
//cc @inikulin, once you approve, I will do the minor changes in the other two repos

**Review by inikulin — CHANGES_REQUESTED** [review]
@diervo a couple of issue left that were not fixed:
https://github.com/HTMLParseErrorWG/html/pull/12#discussion_r114756584
https://github.com/HTMLParseErrorWG/html/pull/12#discussion_r114756846

**Comment by diervo** [comment]
@inikulin Done, sorry for being sloppy

**Comment by inikulin** [comment]
@diervo No worries, sorry for the forced hurry.

**Comment by diervo** [comment]
@inikulin No this is great! If we could merge everything by the end of the week will be awesome. The sooner we finish the early I can do some integrations we are blocked on :)

**Comment by diervo** [comment]
Once you merge this, I will take care of the other two repos, then is only one left.

**Comment by inikulin** [comment]
@diervo Give me an hour, I'm in the middle of rebasing our work on top of the upstream changes.

**Comment by diervo** [comment]
No rush, I have a couple of hours in the afternoon to close whatever is remaining

**Comment by inikulin** [comment]
@diervo This one still not fixed: https://github.com/HTMLParseErrorWG/html/pull/12#discussion_r114756584

File: source
Comment (inikulin) [reply source]: This one still not fixed. Missing space before `<span>parse error</span>` here.

**Review by inikulin — CHANGES_REQUESTED** [review]

**Comment by diervo** [comment]
@inikulin done

**Review by inikulin — APPROVED** [review]

**Pull request merged**
